inference/bot.py

```diff
@@ -85,23 +85,28 @@ def __init__(self, model_name, gpu_id, max_memory):
 
     def do_inference(self, prompt, max_new_tokens, do_sample, temperature, top_k, stream_callback=None):
         stop_criteria = StopWordsCriteria(self._tokenizer, [self.human_id], stream_callback)
-        inputs = (
-            self._tokenizer(prompt, return_tensors='pt')
-            .to(self._model.device)
-        )
-        outputs = self._model.generate(
-            **inputs,
-            max_new_tokens=max_new_tokens,
-            do_sample=do_sample,
-            temperature=temperature,
-            top_k=top_k,
-            pad_token_id=self._tokenizer.eos_token_id,
-            stopping_criteria=StoppingCriteriaList([stop_criteria]),
-        )
-        output = self._tokenizer.batch_decode(outputs)[0]
-
-        # remove the context from the output
-        output = output[len(prompt):]
+        try:
+            inputs = (
+                self._tokenizer(prompt, return_tensors='pt')
+                .to(self._model.device)
+            )
+            outputs = self._model.generate(
+                **inputs,
+                max_new_tokens=max_new_tokens,
+                do_sample=do_sample,
+                temperature=temperature,
+                top_k=top_k,
+                pad_token_id=self._tokenizer.eos_token_id,
+                stopping_criteria=StoppingCriteriaList([stop_criteria]),
+            )
+            del inputs
+            output = self._tokenizer.batch_decode(outputs)[0]
+            del outputs
```

Comment:
Thanks, @MichaMucha. Should `del inputs` and `del outputs` be in the finally block instead?

Reply:
Apologies for the delay-
I suppose in case outputs failed to generate for some reason, they wouldn't be set, therefore `del outputs` in the finally block would throw an exception..

Not sure if a similar case can be made for inputs, but I wanted to avoid the risk of an unhandled exceptio

Reply:
> Not sure if a similar case can be made for inputs, but I wanted to avoid the risk of an unhandled exceptio

Makes sense, 'outputs' isn't defined if it errors out before then. Maybe just move inputs?

Reply:
Can we check if `inputs` and `outputs` are set before `del inputs` and `del outputs` in the finally block?